cpp/test/sparse/sort.cu

```diff
@@ -60,30 +61,26 @@ TEST_P(COOSort, Result)
 
   uniform(h, r, in_vals.data(), params.nnz, float(-1.0), float(1.0));
 
-  int* in_rows_h = (int*)malloc(params.nnz * sizeof(int));
-  int* in_cols_h = (int*)malloc(params.nnz * sizeof(int));
-  int* verify_h  = (int*)malloc(params.nnz * sizeof(int));
+  auto in_rows_h = std::make_unique<int[]>(params.nnz);
```

Comment:
Maybe we could just use a std::vector here?

Reply:
@cjnolet A good point, though I would prefer not to change the test's existing behavior by zero-initializing all the elements of the arrays.  (That would also hinder a sanitizer's ability to track use of uninitialized values.)  It's up to you, though!

Reply:
In general I prefer the use of vectors over smart pointers for the host arrays, especially as we are just performing a loop over the array to directly populate it in the next step.

It's not a huge deal in this test, though, as it's pretty trivial.